src/core/server/services/users/index.ts

```diff
@@ -360,23 +361,88 @@ export async function deactivateToken(
 }
 
 /**
- * updateUsername will update a given User's username.
+ * updateUserUsername will update the current users.
  *
  * @param mongo mongo database to interact with
+ * @param mailer mailer queue instance
  * @param tenant Tenant where the User will be interacted with
  * @param userID the User's ID that we are updating
  * @param username the username that we are setting on the User
  */
-export async function updateUsername(
+export async function updateOwnUsername(
```

Comment:
We've used `updateUsername` to refer to changing ones own username, and `updateUserUsername` as the semantic in other places.